src/core/utils/flash.hpp

```diff
@@ -200,7 +200,7 @@ class Flash : public InstanceLocator
         }
 
     private:
-        static constexpr uint16_t kMaxDataSize = 255;
+        static constexpr uint16_t kMaxDataSize = 512;
```

Comment:
@abtink I had to extend this buffer size to 512B is it ok? Or should I look for a different solution of storing the cert.

Reply:
@abtink Test from Nordics platform show that we currently do not use flash driver but rather settings API.
the code investigation shows https://github.com/openthread/openthread/blob/main/src/core/utils/flash.cpp#L193 that although we are allocating for record a fixed size the real used is written to flash. I don't see any allocations of `Record` other than on stack.

@jwhui maybe you have ideas or comments about this?

There is always an alternative solution we could allocate more than one key for this certificate, and simply split it in chunks.

Reply:
I think the "settings" API does not seem the right place to store an entire Commissioner certificate. It can be up to 1 KB, as defined/discussed today. Maybe more in the future. Enlarging the flash size to 512B would still only fit half a certificate?
Is there another API for more general data blob storage?

Reply:
The OT bottom API does not define any other persistent storage, either we define something new or resolve the issue with settings.  All suggestions are welcomed.

Reply:
IHMO allocating settings multiple keys and dissecting the cert in 256 B chunks would be the best/easiest.

Reply:
Right, so that seems the only API available currently for it. (Is there a restriction in the number of storage keys? Assume we would still have enough keys left.)
Current max size of data is set at 255 - could we set it to 256B ? Then the chunking might be easier 256 iso 255.

@abtink  said a different way would be better, but I guess the only different way would be implementing a sort of blob-store / file-store API that allows larger data items? (e.g. 1KB)
If TCAT is disabled, the file-store should not take up any flash space then.

@arnulfrupp FYI!

Reply:
Increasing the maximum record size should be OK. Of course, with the given implementation, it will increase memory pressure of the flash module.

That said, it's better if the platform can directly provide a platform-optimized implementation for `otPlatSettings*` APIs.

Reply:
Ok so let's extend it to 1K (as proposed also in https://threadgroup.atlassian.net/browse/BHC-744) I can do it conditionally with some ifdef.

Reply:
@canisLupus1313 I think the `kMaxDataSize = 1024;` needs to be wrapped in an #ifdef still?
We can use `kCommissionerCertMaxLength ` instead of plain 1024 there, in case TCAT is enabled.

If TCAT is disabled, then use 255 like before (noting @jwhui  did mention a trade-off between max size and stack size so we probably don't need 1024 for non-TCAT devices). From what I can see this is the only item left to address.